lib/webauthn/credential.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "webauthn/authenticator_assertion_response"
+require "webauthn/authenticator_attestation_response"
+require "webauthn/client_utils"
+
+module WebAuthn
+  class Credential
+    VALID_TYPE = "public-key"
+
+    def self.from_json(json)
+      if json["type"] != VALID_TYPE
+        raise "invalid credential type"
+      end
+
+      id = WebAuthn::ClientUtils.decode(json["id"])
+      response_json = json["response"]
+
+      response =
+        if response_json["attestationObject"]
+          AuthenticatorAttestationResponse.from_json(response_json)
+        elsif response_json["signature"]
+          AuthenticatorAssertionResponse.from_json(response_json, id)
+        else
+          raise "invalid response"
```

Comment:
Might be better to split `.from_json` into two more context aware methods called `.from_create` and `.from_get`, than to try guess here were the json is coming from.

Reply:
Done here https://github.com/cedarcode/webauthn-ruby/pull/224